gnovm/pkg/gnolang/machine.go

```diff
@@ -1154,127 +1154,130 @@ func (m *Machine) incrCPU(cycles int64) {
 }
 
 const (
+	// CPU cycles
```

Comment:
Can we add a reference to the benchmarks and methodology?

Reply:
I'm concerned that this approach renders `ugnot` nearly valueless, as it results in excessively long and difficult-to-read numbers. 

We should either adopt a strategy that sacrifices some precision, such as dividing by N (which could be a parameter), or consider introducing another level of `gnot` conversion, like `ngnot` for gas, to maintain the relevance of `ugnot`.

Reply:
@moul @zivkovicmilos Totally agree, the large number of gas and fees is not a good user experience.  Let me think about how we can fix it later.

Reply:
> Can we add a reference to the benchmarks and methodology?

I will write a separate article for the gas benchmarking.  It is related to PR https://github.com/gnolang/gno/pull/2241